src/shared/dissect-image.c

```diff
@@ -4421,6 +4421,7 @@ int verity_dissect_and_mount(
                         r = extension_release_validate(
                                         dissected_image->image_name,
                                         extension_release_data->os_release_id,
+                                        /* host_os_release_id_like = */ NULL,
```

Comment:
please implement this too, otherwise ExtensionDirectories will be able to use ID_LIKE, but ExtensionImages won't - it should be simple to do, just add the parameter to ExtensionReleaseData